lib/logstash/outputs/elasticsearch_java.rb

```diff
@@ -61,14 +64,14 @@
 # - Events from the retry queue are submitted again either when the queue reaches its max size or when
 #   the max interval time is reached, which is set in :retry_max_interval.
 # - Events which are not retryable or have reached their max retry count are logged to stderr.
-class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
+class LogStash::Outputs::ElasticSearchJava < LogStash::Outputs::Base
   attr_reader :client
 
   include Stud::Buffer
-  RETRYABLE_CODES = [429, 503]
+  RETRYABLE_CODES = [409, 429, 503]
```

Comment:
Yes it is retryable when you have a lots of update on the same document in different bulk !
But we can delegate this to elasticsearch (update api gives a parameter 'retry_on_conflict'). I didn't know this feature when I wrote this code !